.github/workflows/update-notices.yml

```diff
@@ -0,0 +1,92 @@
+name: Update Local Notice Files
+# This action can be run on-demand against a branch or scheduled every day.
+# It updates the checked-in TPN files and creates a PR with the changes.
+
+on:
+  workflow_dispatch:
+  schedule:
+   - 0 9 * * *
+
+jobs:
+  main:
+    name: Update Local Notice Files
+
+    # has to be windows because of the windows installer project
+    runs-on: windows-latest
+
+    env:
+      DOTNET_NOLOGO: true
+      DOTNET_CLI_TELEMETRY_OPTOUT: true
+      GenerateNoticeUseLocalFile: false
+      GenerateNoticeUpdateLocalFile: true
+
+    steps:
```

Comment:
>steps [](http://example.com/codeflow?start=4&length=5)

There isn't a good way to reuse the build logic without refactoring into scripts. We'll do that at some point...